### DISCUSSION TITLE
Project 3

### DISCUSSION THREAD

**Discussion by awc17002** [cooperrc/computational-mechanics#119] [General]
For part 2 of the project, we have to integrate the rocket function implicitly and explicitly. I'm trying to use Heun's method for the implicit integration and RK2 (Modified Euler's) for the explicit integration. My question is how would I call a function like Heun's to call the Rocket function in order to graph it and show it converges.

**Comment by m-c-johnston** [comment]
To use the Heun (or RK2) method I set my initial values and created an array of zeros, N rows and 3 columns.
Then set each initial value to the first row of its respective column (height, velocity, mass).
Finally, use a for loop the work through the function;  
for i in range(N-1):
    heun[i+1] = heun_step(heun[i], function, dt).
Then just plot all the rows from the height and mass columns.
